src/main/java/uk/gov/ons/census/casesvc/service/BlankQuestionnaireService.java

```diff
@@ -0,0 +1,178 @@
+        new BlankQuestionnaireService.NoActionRequired());
+    rules.put(
+        new BlankQuestionnaireService.Key("SPG", "U", CONT_FORM_TYPE, false),
+        new BlankQuestionnaireService.NoActionRequired());
+    rules.put(
+        new BlankQuestionnaireService.Key("SPG", "E", CONT_FORM_TYPE, false),
+        new BlankQuestionnaireService.NoActionRequired());
+    rules.put(
+        new BlankQuestionnaireService.Key("CE", "U", CONT_FORM_TYPE, false),
+        new BlankQuestionnaireService.NoActionRequired());
+    rules.put(
+        new BlankQuestionnaireService.Key("CE", "E", CONT_FORM_TYPE, false),
+        new BlankQuestionnaireService.NoActionRequired());
+    rules.put(
+        new BlankQuestionnaireService.Key("HH", "U", CONT_FORM_TYPE, true),
+        new BlankQuestionnaireService.NoActionRequired());
+    rules.put(
+        new BlankQuestionnaireService.Key("SPG", "U", CONT_FORM_TYPE, true),
+        new BlankQuestionnaireService.NoActionRequired());
+    rules.put(
+        new BlankQuestionnaireService.Key("SPG", "E", CONT_FORM_TYPE, true),
+        new BlankQuestionnaireService.NoActionRequired());
+    rules.put(
+        new BlankQuestionnaireService.Key("CE", "U", CONT_FORM_TYPE, true),
+        new BlankQuestionnaireService.NoActionRequired());
+    rules.put(
+        new BlankQuestionnaireService.Key("CE", "E", CONT_FORM_TYPE, true),
+        new BlankQuestionnaireService.NoActionRequired());
+  }
+
+  public void handleBlankQuestionnaire(
+      Case caze, UacQidLink uacQidLink, EventTypeDTO causeEventType) {
+    BlankQuestionnaireService.Key ruleKey = makeRulesKey(caze, uacQidLink);
+
+    if (!rules.containsKey(ruleKey)) {
+      throw new RuntimeException(ruleKey.toString() + " does not map to any known processing rule");
+    }
+
+    BlankQreRule rule = rules.get(ruleKey);
+    rule.run(caze, causeEventType);
+  }
+
+  private BlankQuestionnaireService.Key makeRulesKey(Case caze, UacQidLink uacQidLink) {
+    String formType = mapQuestionnaireTypeToFormType(uacQidLink.getQid());
+
+    return new BlankQuestionnaireService.Key(
+        caze.getCaseType(),
+        caze.getAddressLevel(),
+        formType,
+        caseHasOtherValidReceiptForFormType(caze, formType, uacQidLink));
+  }
+
+  private boolean caseHasOtherValidReceiptForFormType(
+      Case caze, String formType, UacQidLink uacQidLink) {
+    return caze.getUacQidLinks().stream()
+        .anyMatch(
+            u ->
+                !u.isActive()
+                    && !u.isBlankQuestionnaire()
+                    && Objects.equals(mapQuestionnaireTypeToFormType(u.getQid()), formType)
+                    && !u.equals(uacQidLink));
+  }
+
+  @AllArgsConstructor
+  @EqualsAndHashCode
+  @ToString
+  private class Key {
+
+    private String caseType;
+    private String addressLevel;
+    private String formType;
+    private boolean hasOtherValidReceiptForFormType;
+  }
+
+  private interface BlankQreRule {
```

Comment:
not sure why this has been abbreviated. We can use fully descriptive names for things - there's enough space

Reply:
I was probably just tired of typing `Questionnaire`, will fix for consistency!